lib/packwerk.rb

```diff
@@ -14,8 +14,11 @@ module Packwerk
   extend ActiveSupport::Autoload
 
   # Public APIs
+  autoload :ApplicationLoadPaths
   autoload :Checker
   autoload :Cli
+  autoload :ConstantDiscovery
```

Comment:
Needed to build a `Packwerk::Reference`, existing public API.

Reply:
I assume that's because we need `ConstantContext`? Should we just move this toplevel? We don't have a valid use-case of using it in an extension, do we?

Reply:
Yes, we only use `ConstantContext`. That makes sense – I've moved it to the top-level so it can be public without all of `ConstantDiscovery` being public.